### PR TITLE
Add README for comms

### LINKED ISSUE #24: Add nice README

### PR DESCRIPTION
**Pull request by iasiu**
* Add README with examples and explanations

* Update changelog and pubspec

Closes #24

### CODE REVIEW

File: packages/comms/README.md [hunk 1]
```diff
@@ -1,3 +1,204 @@
...
+
+  // At the end don't forget to close the connection and cleanup.
+  lightBulb.dispose();
+}
+```
+
+### Multiple Listeners and Senders
+
+With `comms` any `Sender` will send a message to any `Listener` sharing the same
+message type.
```
Comment (bartekpacia) [inline packages/comms/README.md:86]:
```suggestion

A `Sender` sends a message to all `Listener` sharing the same
message type.
```

[See this](https://en.wikiversity.org/wiki/Technical_writing/Style#Understanding_present_tense)

Also, I'd add one more sentence with some example:

```
For example, if you have a single `Sender<UserData>`, and 3 `Listener<UserData`, then whenever that sender sends a message, all the listeners will receive the message (simultaneously? idk).
```

File: packages/comms/README.md [hunk 2]
```diff
@@ -1,3 +1,204 @@
...
+```
+
+## Custom Senders
+To create a custom `Sender` for example to send multiple message types, you
+ can use `getSend` which returns `send` method for messages of passed type.
+
+```dart
+mixin CustomSender {
+  final sendString = getSend<String>();
+  final sendInt = getSend<int>();
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply packages/comms/README.md:184]: On the other hand, this is a mini tutorial, so maybe simple types make more sense.

I'm torn.

File: packages/comms/README.md [hunk 3]
```diff
@@ -1,3 +1,204 @@
...
+
+  void sendMessages() {
+    sendString('hello');
+    sendInt(1);
+  }
+}
+```
+
+You can also use `getSend` to send messages from anywhere without even
```
Comment (bartekpacia) [inline packages/comms/README.md:193]:
```suggestion
You can also use the `getSend()` function to send messages from anywhere without even
```

File: packages/comms/README.md [hunk 4]
```diff
@@ -1,3 +1,204 @@
...
+  // handle connection between them.
+  final lightBulb = LightBulb();
+  final lightSwitch = LightSwitch();
+
+  print(lightBulb.enabled); // false
+
+  lightSwitch.enable();
+    
+  // Because communication is asynchronuos we have to wait
+  // just a little bit for the message to reach the `lightBulb`.
```
[the remark this answers is not in the document]
Comment (iasiu) [reply packages/comms/README.md:73]: well yeah, but I don't want to go into details of asynchrony in dart here, and just explain why if anyone copied this code, it wouldn't work without this line, if you have any ideas I am open to changing it

File: packages/comms/README.md [hunk 5]
```diff
@@ -1,3 +1,204 @@
...
+  lightSwitchB.enable();
+
+  print(lightBulbA.enabled); // true
+  print(lightBulbB.enabled); // true
+}
+```
+
+## Handling initial message
+To handle the last message sent before creating an instance of `Listener` you
+can use `onInitialMessage` which will be invoked after `Listener` calls `listen`
```
[the remark this answers is not in the document]
Comment (iasiu) [reply packages/comms/README.md:108]: That's ok but personally I don't like adding ()

File: packages/comms/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (iasiu) [reply packages/comms/README.md:184]: I don't want to advise anyone on how to use this package, also I think that `Multiple Listeners and Senders` section kind of means that

[the remark this answers is not in the document]
Comment (iasiu) [reply packages/comms/README.md:184]: Also in this example I wanted to show that you can easily use both functions without showing custom types definitions

File: packages/comms/README.md [hunk 6]
```diff
@@ -1,3 +1,209 @@
 # comms
 
-Simple communication pattern abstraction on streams, created for communication between logic classes.
+[![codestyle][pub_badge_style]][pub_badge_link]
+
+`comms` is a simple communication pattern abstraction on streams, created for
+communication between any classes. It allows `Listener`s to easily react to
+messages sent by `Sender`s.
+
+For use in Flutter projects, check out [flutter_comms].
```
Comment (bartekpacia) [inline packages/comms/README.md:9]: It'd be useful to explain what benefits `flutter_comms` brings – e.g automatic `dispose()`

Comment (iasiu) [reply packages/comms/README.md:9]: I'd expect to see it there in readme, first line or sth

Comment (bartekpacia) [reply packages/comms/README.md:9]: ah right, makes sense

Comment (bartekpacia) [reply packages/comms/README.md:9]: since you're on writing docs frenzy right now, why not write a cool readme for `flutter_comms` now as well? 👅

Comment (iasiu) [reply packages/comms/README.md:9]: I need a break 😄

File: packages/comms/README.md [hunk 7]
```diff
@@ -1,3 +1,209 @@
 # comms
 
-Simple communication pattern abstraction on streams, created for communication between logic classes.
+[![codestyle][pub_badge_style]][pub_badge_link]
```
Comment (bartekpacia) [inline packages/comms/README.md:3]: add badge with a link to pub.dev

[example here](https://raw.githubusercontent.com/leancodepl/patrol/master/README.md)

Comment (iasiu) [reply packages/comms/README.md:3]: I'll add it to the main readme later

**Pull request merged**